script.js

```diff
@@ -0,0 +1,245 @@
+const allButtons = document.getElementsByTagName("button");
+const numberButtons = document.querySelectorAll("[data-number]");
+const operationButtons = document.querySelectorAll("[data-operation]");
+const equalsButton = document.querySelector("[data-equals]");
+const clearButton = document.querySelector("[data-clear]");
+const deleteButton = document.querySelector("[data-delete]");
+const plusminus = document.querySelector("[data-plusminus]");
+const inputNumberField = document.querySelector(".inputNumber");
+
+let inputOperator = undefined;
+let savedInputNumber = undefined;
+let currentNumber = 0;
+let resultNumber = undefined;
+
+let replaceNumber = true;
+
+document.body.onkeypress = (e) => {
+  if (isDigit(e.key)) {
+    checkInput(e.key);
+    keyPressEvent(e.key);
+  }
+};
+
+document.body.onkeydown = (e) => {
+  switch (e.key.toLowerCase()) {
+    case "enter":
```

Comment:
this is potentially doing more than just calculating when the focus is on other buttons.